### PR TITLE
[#58] disable RSpec/LetSetup cop by default

### LINKED ISSUE #58: Rspec: RSpec/LetSetup - not to use `let!`
[This cop](https://rubocop-rspec.readthedocs.io/en/latest/cops_rspec/#rspecletsetup) does not allow to use`let!` for setting up test data. 

By documentation, it should be disabled by default but I caught this issue in the project.

```
spec/services/review_status/move_to_published_spec.rb:14:5: C: RSpec/LetSetup: Do not use let! for test setup.
    let!(:status) { create :review_status, workflow: workflow, status: :approved, created_by: user }
    ^^^^^^^^^^^^^
spec/services/review_status/move_to_published_spec.rb:34:7: C: RSpec/LetSetup: Do not use let! for test setup.
      let!(:status) { create :review_status, workflow: workflow, status: :pending, created_by: user }
      ^^^^^^^^^^^^^
```

Any thoughts? I didn't investigate why it was added to the default config so I don't have any thoughts now.

### PR DESCRIPTION
**Pull request by lazycoder9**
Before you submit a pull request, please make sure you have to follow:

- [x] read and know items from the [Contributing Guide](CONTRIBUTING.md#pull-requests)
- [x] add a description of the problem you're trying to solve (short summary from related issue)
- [x] verified that cops are ordered by alphabet
- [x] add a note to the style guide docs (if it needs)
- [x] add a note to the changelog file
- [x] the commit message contains the number of the related issue (if it presents)
  and word `Fix` if this PR closes related issue
- [x] squash all commits before submitting to review

#58

### CODE REVIEW

**Comment by roman-dubrovsky** [comment]
@lazycoder9 if you want we can merge this commit but (w/o word close) (so that the issue won't be closed)

**Comment by lazycoder9** [comment]
@roman-dubrovsky works for me)

**Comment by lazycoder9** [comment]
@roman-dubrovsky squashed and fixed commit and PR messages, is it enough to avoid closing the issue?

File: doc/STYLE_GUIDE_RSPEC.md [hunk 1]
```diff
@@ -153,3 +153,16 @@ end
...
+  We allow using `let!` in your test
+  <sup>[[link](#rspec-let-setup)]</sup>
+
+```ruby
+# ok
+let!(:my_widget) { create(:widget) }
+
+it 'counts widgets' do
+  expect(Widget.count).to eq(1)
+end
```
Comment (roman-dubrovsky) [inline doc/STYLE_GUIDE_RSPEC.md]:
```ruby
# better

before do
  create(:widget)
end

it "counts widgets" do
  expect(Widget.count).to eq(1)
end 

# good

let!(:my_widget) { create(:widget) }

it "returns the last widget" do
  expect(Widget.last).to eq my_widget
end
```

File: doc/STYLE_GUIDE_RSPEC.md [hunk 2]
```diff
@@ -153,3 +153,16 @@ end
 ```
 
 **Note:** this is a Part of Rspec configuration. Read more [by link](https://github.com/rubocop-hq/rubocop-rspec#enforcing-an-explicit-rspec-receiver-for-top-level-methods-disabling-monkey-patching).
+
+* <a name="rspec-let-setup"></a>
+  We allow using `let!` in your test
```
Comment (roman-dubrovsky) [inline doc/STYLE_GUIDE_RSPEC.md]: We allow using `let!` in your test, but suggest not to use it for setting up testing data

File: doc/STYLE_GUIDE_RSPEC.md [hunk 3]
```diff
@@ -153,3 +153,16 @@ end
...
+  <sup>[[link](#rspec-let-setup)]</sup>
+
+```ruby
+# ok
+let!(:my_widget) { create(:widget) }
+
+it 'counts widgets' do
```
Comment (roman-dubrovsky) [inline doc/STYLE_GUIDE_RSPEC.md]: use double quotes

File: doc/STYLE_GUIDE_RSPEC.md [hunk 4]
```diff
@@ -153,3 +153,27 @@ end
 ```
 
 **Note:** this is a Part of Rspec configuration. Read more [by link](https://github.com/rubocop-hq/rubocop-rspec#enforcing-an-explicit-rspec-receiver-for-top-level-methods-disabling-monkey-patching).
+
+* <a name="rspec-let-setup"></a>
+  We allow using let! in your test, but suggest not to use it for setting up testing data
```
Comment (roman-dubrovsky) [inline doc/STYLE_GUIDE_RSPEC.md]: dot in the end

**Pull request merged**